testbed/testbed/load_generator.go

```diff
@@ -210,22 +208,41 @@ func (ps *ProviderSender) generate() {
 			defer workers.Done()
 			t := time.NewTicker(time.Second / time.Duration(ps.options.DataItemsPerSecond/ps.options.ItemsPerBatch/numWorkers))
 			defer t.Stop()
+
+			var (
+				prevErr error
+				err     error
+				b       bool
+			)
 			for {
 				select {
 				case <-t.C:
 					switch ps.Sender.(type) {
 					case TraceDataSender:
-						ps.generateTrace()
+						b, err = ps.generateTrace()
 					case MetricDataSender:
-						ps.generateMetrics()
+						b, err = ps.generateMetrics()
 					case LogDataSender:
-						ps.generateLog()
+						b, err = ps.generateLog()
 					default:
-						log.Printf("Invalid type of ProviderSender sender")
+						log.Printf("Invalid type of ProviderSender sender, quiting load generation")
+						continue
```

Comment:
Why `continue` here instead of `return`? It seems to contradict the log message, and results in just coming back to the same case in the `select` from what I can tell, resulting in log spam without reason.

I'm not sure how returning here would impact test runs, but hopefully the failure reason would be obvious enough with the log message.

Reply:
> and results in just coming back to the same case in the select from what I can tell, resulting in log spam without reason

I agree. The only reason I went with the `continue` was to keep the same behavior as before. I think we should just error if the signal type is unsupported in the [constructor itself](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/2daff18c2e3d9a1b63df72fd5c26f8c83a2d0683/testbed/testbed/load_generator.go#L98), WDYT?

Reply:
I'd prefer to error sooner rather than later since this is for testing. No reason to waste time if it's going to fail. 👍

Reply:
Updated the logic to error in the constructor if the sender type is not supported.